### PR TITLE
Silent retry if conflict

### PR DESCRIPTION
**Pull request by miiu96**
- Added retry mechanism in case of a response with `code 409`

### CODE REVIEW

File: factory/indexerFactory.go [hunk 1]
```diff
@@ -72,10 +78,16 @@ func NewIndexer(args *ArgsIndexerFactory) (indexer.Indexer, error) {
...
-		Addresses: []string{args.Url},
-		Username:  args.UserName,
-		Password:  args.Password,
-		Logger:    &logging.CustomLogger{},
+		Addresses:     []string{args.Url},
+		Username:      args.UserName,
+		Password:      args.Password,
+		Logger:        &logging.CustomLogger{},
+		RetryOnStatus: []int{http.StatusConflict},
+		RetryBackoff: func(attempt int) time.Duration {
```
Comment (iulianpascalau) [inline factory/indexerFactory.go:86]: this is a little bit ugly and very hard to test.
Suggestion here is to create an interface with one or 2 implementations and proper unit tests

**Review by iulianpascalau — CHANGES_REQUESTED** [review]

File: factory/indexerFactory.go (hunk 1 above)
Comment (bogdan-rosianu) [reply factory/indexerFactory.go:86]: I don't think it's worth. Elasticsearch libs expect that argument as  `func(attempt int) time.Duration` . I don't think it is necessary that we create a new interface that exposes a `RetryFunction func(attempt int) time.Duration`

Comment (iulianpascalau) [reply factory/indexerFactory.go:86]: It is pretty easy to be implemented. Will create a PR for this

**Review by iulianpascalau — APPROVED** [review]

File: factory/indexerFactory.go (hunk 1 above)
Comment (miiu96) [reply factory/indexerFactory.go]: done

**Pull request merged**